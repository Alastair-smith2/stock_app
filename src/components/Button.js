import React, { Component } from "react";
import { View, Text, TouchableHighlight, StyleSheet } from "react-native";

export default class Button extends Component {
  constructor(props) {
    super(props);

    this.onPress = this.onPress.bind(this);
  }
  onPress = () => {
    this.props.buttonPress();
  };

  render() {
    return (
      <TouchableHighlight
        onPress={this.onPress}
        style={[styles.container, this.props.buttonStyle]}
      >
        <Text style={styles.label}>{this.props.title}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  label: {
    fontSize: 20,
    color: "#fff"
  }
});
