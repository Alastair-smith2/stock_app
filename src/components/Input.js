import React, { Component } from "react";
import { View, Text, TextInput, StyleSheet } from "react-native";

export default class Input extends Component {
  onTextChange = props => {
    this.props.onTextInput(this.props.type, props);
  };

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.label}>{this.props.label}</Text>
        <TextInput
          value={this.props.value}
          placeholder={`Please enter you ${this.props.type}`}
          onChangeText={text => this.onTextChange(text)}
          type={this.props.type}
          underlineColorAndroid={"transparent"}
          autoCapitalize={"none"}
          style={styles.textInput}
        />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignContent: "center"
  },
  label: {
    fontSize: 16,
    fontFamily: "Futura"
  },
  textInput: {
    flex: 1.5,
    height: 40,
    borderColor: "#BBB",
    borderWidth: 1,
    padding: 10
  }
});
