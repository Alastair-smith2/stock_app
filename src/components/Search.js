import React, { Component } from "react";
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  TouchableHighlight
} from "react-native";
import Button from "./Button";
import Input from "./Input";

export default class Search extends Component {
  constructor(props) {
    super(props);

    this.searchSubmit = this.searchSubmit.bind(this);
    this.onTextChange = this.onTextChange.bind(this);
  }
  onTextChange = (type, props) => {
    this.props.onTextSearch(type, props);
  };

  searchSubmit = () => {
    this.props.submitSearch();
  };

  render() {
    return (
      <View style={styles.container}>
        {this.props.searchType == "currency" ? (
          <View style={styles.container}>
            <View style={styles.rowContainer}>
              <View style={styles.inputContainer}>
                <Input
                  value={this.props.currencyOne}
                  onTextInput={this.onTextChange}
                  type={this.props.typeOne}
                  label={this.props.labelOne}
                />
              </View>
              <View style={styles.padding} />
              <View style={styles.inputContainer}>
                <Input
                  value={this.props.currencyTwo}
                  onTextInput={this.onTextChange}
                  type={this.props.typeTwo}
                  label={this.props.labelTwo}
                />
              </View>
            </View>
            <View style={styles.buttonContainer}>
              <View style={styles.alternative}>
                <Button
                  title={"Exchange"}
                  buttonStyle={styles.button}
                  buttonPress={this.searchSubmit}
                />
              </View>
            </View>
          </View>
        ) : (
          <View style={styles.container}>
            <View style={styles.stockContainer}>
              <Input
                value={this.props.value}
                onTextInput={this.onTextChange}
                type={this.props.type}
                label={this.props.label}
              />
            </View>
            <View style={styles.buttonContainer}>
              <View style={styles.alternative}>
                <Button
                  buttonStyle={styles.button}
                  buttonPress={this.searchSubmit}
                  title={"Search"}
                />
              </View>
            </View>
          </View>
        )}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  rowContainer: {
    flex: 0.6,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    marginBottom: 10
  },
  stockContainer: {
    flex: 0.6,
    marginBottom: 10
  },
  inputContainer: {
    flex: 1
  },
  padding: {
    flex: 0.05
  },
  alternative: {
    flex: 0.5,
    flexDirection: "row",
    justifyContent: "center"
  },
  buttonContainer: {
    flex: 1
  },
  button: {
    backgroundColor: "#EA572D",
    justifyContent: "center",
    alignItems: "center",
    flex: 0.5,
    borderRadius: 60
  }
});
