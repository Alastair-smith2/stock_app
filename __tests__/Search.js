import React from "react";
import { shallow } from "enzyme";

import Search from "../src/components/Search.js";
import Input from "../src/components/Input.js";
import Button from "../src/components/Button.js";

describe("The Search component", () => {
  let wrapper;
  let onTextSearch = jest.fn();
  let onSearchSubmit = jest.fn();
  beforeEach(() => {
    wrapper = shallow(
      <Search
        searchType={"stock"}
        value={""}
        label={"Stock symbol"}
        type="search"
        submitSearch={onSearchSubmit}
        onTextSearch={onTextSearch}
      />
    );
    onTextSearch.mockReset();
    onSearchSubmit.mockReset();
    wrapper.setProps({ error: false });
  });
  describe("It should render", () => {
    it("Should render", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Features", () => {
    it("Should have 1 Input component", () => {
      expect(wrapper.find(Input)).toHaveLength(1);
    });

    it("Should have 1 Button component", () => {
      expect(wrapper.find(Button)).toHaveLength(1);
    });

    it("Should have a function that handles the Search change of TextSearch", () => {
      wrapper
        .find(Input)
        .props()
        .onTextInput("Alastair");
      expect(onTextSearch.mock.calls.length).toEqual(1);
    });

    it("Should have a prop called onTextSearch that is called via the component's onTextChange function", () => {
      wrapper.instance().onTextChange("Alastair");
      expect(onTextSearch.mock.calls.length).toEqual(1);
      expect(onTextSearch.mock.calls[0][0]).toEqual("Alastair");
    });

    it("Should be able to handle a search function", () => {
      wrapper
        .find(Button)
        .props()
        .buttonPress();
      expect(onSearchSubmit.mock.calls.length).toEqual(1);
    });

    it("Should change depending upon the search type", () => {
      wrapper.setProps({ searchType: "currency" });
      // only with the currency prop will there be two inputs
      expect(wrapper.find(Input)).toHaveLength(2);
    });
  });
});
