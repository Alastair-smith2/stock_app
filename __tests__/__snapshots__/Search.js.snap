// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search component It should render Should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    error={false}
    label="Stock symbol"
    onTextSearch={[MockFunction]}
    searchType="stock"
    submitSearch={[MockFunction]}
    type="search"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0.6,
              "marginBottom": 10,
            }
          }
        >
          <Input
            label="Stock symbol"
            onTextInput={[Function]}
            placeholder={undefined}
            type="search"
            value=""
          />
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.5,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Button
              buttonPress={[Function]}
              buttonStyle={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA572D",
                  "borderRadius": 60,
                  "flex": 0.5,
                  "justifyContent": "center",
                }
              }
              title="Search"
            />
          </View>
        </View>
      </View>,
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 0.6,
                "marginBottom": 10,
              }
            }
          >
            <Input
              label="Stock symbol"
              onTextInput={[Function]}
              placeholder={undefined}
              type="search"
              value=""
            />
          </View>,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0.5,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Button
                buttonPress={[Function]}
                buttonStyle={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EA572D",
                    "borderRadius": 60,
                    "flex": 0.5,
                    "justifyContent": "center",
                  }
                }
                title="Search"
              />
            </View>
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              label="Stock symbol"
              onTextInput={[Function]}
              placeholder={undefined}
              type="search"
              value=""
            />,
            "style": Object {
              "flex": 0.6,
              "marginBottom": 10,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "label": "Stock symbol",
              "onTextInput": [Function],
              "placeholder": undefined,
              "type": "search",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <View
              style={
                Object {
                  "flex": 0.5,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Button
                buttonPress={[Function]}
                buttonStyle={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EA572D",
                    "borderRadius": 60,
                    "flex": 0.5,
                    "justifyContent": "center",
                  }
                }
                title="Search"
              />
            </View>,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                buttonPress={[Function]}
                buttonStyle={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EA572D",
                    "borderRadius": 60,
                    "flex": 0.5,
                    "justifyContent": "center",
                  }
                }
                title="Search"
              />,
              "style": Object {
                "flex": 0.5,
                "flexDirection": "row",
                "justifyContent": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "buttonPress": [Function],
                "buttonStyle": Object {
                  "alignItems": "center",
                  "backgroundColor": "#EA572D",
                  "borderRadius": 60,
                  "flex": 0.5,
                  "justifyContent": "center",
                },
                "title": "Search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "flex": 0.6,
                "marginBottom": 10,
              }
            }
          >
            <Input
              label="Stock symbol"
              onTextInput={[Function]}
              placeholder={undefined}
              type="search"
              value=""
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 0.5,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Button
                buttonPress={[Function]}
                buttonStyle={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#EA572D",
                    "borderRadius": 60,
                    "flex": 0.5,
                    "justifyContent": "center",
                  }
                }
                title="Search"
              />
            </View>
          </View>
        </View>,
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "flex": 0.6,
                  "marginBottom": 10,
                }
              }
            >
              <Input
                label="Stock symbol"
                onTextInput={[Function]}
                placeholder={undefined}
                type="search"
                value=""
              />
            </View>,
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 0.5,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Button
                  buttonPress={[Function]}
                  buttonStyle={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#EA572D",
                      "borderRadius": 60,
                      "flex": 0.5,
                      "justifyContent": "center",
                    }
                  }
                  title="Search"
                />
              </View>
            </View>,
          ],
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Input
                label="Stock symbol"
                onTextInput={[Function]}
                placeholder={undefined}
                type="search"
                value=""
              />,
              "style": Object {
                "flex": 0.6,
                "marginBottom": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Stock symbol",
                "onTextInput": [Function],
                "placeholder": undefined,
                "type": "search",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <View
                style={
                  Object {
                    "flex": 0.5,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Button
                  buttonPress={[Function]}
                  buttonStyle={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#EA572D",
                      "borderRadius": 60,
                      "flex": 0.5,
                      "justifyContent": "center",
                    }
                  }
                  title="Search"
                />
              </View>,
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Button
                  buttonPress={[Function]}
                  buttonStyle={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#EA572D",
                      "borderRadius": 60,
                      "flex": 0.5,
                      "justifyContent": "center",
                    }
                  }
                  title="Search"
                />,
                "style": Object {
                  "flex": 0.5,
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonPress": [Function],
                  "buttonStyle": Object {
                    "alignItems": "center",
                    "backgroundColor": "#EA572D",
                    "borderRadius": 60,
                    "flex": 0.5,
                    "justifyContent": "center",
                  },
                  "title": "Search",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
